From fd2a9555fbf95ba312268552b11c1fe08478a6b6 Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Mon, 21 Oct 2024 11:57:58 +0200
Subject: [PATCH 1307/1695] drivers: stm32_bsec3: map bsec_mirror non secure if
 M33 is TDCID

If Cortex-M33 is TDCID, it will set bsec_mirror access to non-secure
only. Adapt the MMU setting for this area in this case.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git be232d3c35dc1730a939b65eb260e376b16be0e6 drivers: stm32_bsec3: map bsec_mirror non secure if M33 is TDCID]
Change-Id: I9a1bf66ddc1cd75c4ee4d56bb70ef2d70e87591a
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/412909
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/drivers/stm32_bsec3.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/core/drivers/stm32_bsec3.c b/core/drivers/stm32_bsec3.c
index 72b5f93ce..5d6ff85ad 100644
--- a/core/drivers/stm32_bsec3.c
+++ b/core/drivers/stm32_bsec3.c
@@ -1220,7 +1220,11 @@ static TEE_Result initialize_bsec(void)
 
 	plat_bsec_get_static_cfg(&cfg);
 
-	va = core_mmu_get_va(cfg.shadow, MEM_AREA_RAM_SEC, SIZE_4K);
+	if (IS_ENABLED(CFG_STM32_CM33TDCID))
+		va = core_mmu_get_va(cfg.shadow, MEM_AREA_RAM_NSEC, SIZE_4K);
+	else
+		va = core_mmu_get_va(cfg.shadow, MEM_AREA_RAM_SEC, SIZE_4K);
+
 	if (!va)
 		return TEE_ERROR_GENERIC;
 
-- 
2.47.2

